File: app/src/main/java/mb00/android/codehub/ui/repo/inject/RepoLicenseFragmentModule.java
package mb00.android.codehub.ui.repo.inject;

import dagger.Module;
import dagger.Provides;
import mb00.android.codehub.ui.repo.interactor.RepoInteractor;
import mb00.android.codehub.ui.repo.router.RepoRouter;
import mb00.android.codehub.ui.repo.viewmodel.RepoLicenseViewModel;


@Module
public class RepoLicenseFragmentModule {

    @Provides
    RepoLicenseViewModel provideRepoLicenseViewModel(RepoInteractor interactor, RepoRouter router) {
        return new RepoLicenseViewModel(interactor, router);
    }

}

File: app/src/main/java/mb00/android/codehub/ui/repo/inject/RepoLicenseFragmentModule.kt
package mb00.android.codehub.ui.repo.inject

import dagger.Module
import dagger.Provides
import mb00.android.codehub.ui.repo.interactor.RepoInteractor
import mb00.android.codehub.ui.repo.router.RepoRouter
import mb00.android.codehub.ui.repo.viewmodel.RepoLicenseViewModel


@Module
class RepoLicenseFragmentModule {

    @Provides
    internal fun provideRepoLicenseViewModel(interactor: RepoInteractor, router: RepoRouter): RepoLicenseViewModel {
        return RepoLicenseViewModel(interactor, router)
    }

}
